deid/dicom/parser.py

```diff
@@ -269,6 +269,21 @@ def skip(self):
             skips = self.config.get("get", {}).get("skip", {})
         return skips
 
+    @property
+    def keep(self):
+        """
+        Return a list of fields to keep, as defined by all KEEP actions in recipe
+        """
+        keeps = []
+        if self.recipe.deid is not None:
+            keeps = list(
```

Comment:
Generally it's faster to use [] or {} as opposed to the types, and I suspect this is considered Pythonic but it's not easy for the reader to understand. Could you please maintain the same structure you used in the parse function before?

Reply:
Previously I iterated all actions, adding them by one, and it was one-time iteration. Now potentially property can be called multiple times if some new code is added, so using `append` makes no sense (no point to append to already populated). 
I can get rid of lambda, but anyway it should be `keeps = <new array>`, not previously used `append`.
So it would be 
```
keeps =[action.get('field') for action in self.recipe.get_actions(action='KEEP')]
```
As pythonic as it could be and no lambdas :)
Or did you mean smth else?

Reply:
Yep that looks good! I would add a check like "if action and action.get('field')` to make sure the action is defined and it has a field.

Reply:
what could cause it to be undefined? `get_actions` returns empty array even if recipe has no KEEP actions, checked it. But how can it return i.e `[None]`?

Reply:
Nope, it causes other failure in `parser.py:355`, `if re.search("^values", field):`. According to docs `field` is `if provided`, but actually if it is `None` it raises exception.

Reply:
In that sense `pydicom` tries to emulate `dict.get()` in a bad manner - Python's `dict` fails if key not present and `default` not given, but `pydicom` defaults to `default=None`. Meaning `perform_action` will be called with `field == None` if it is missing in recipe, and exception will be thrown.

Reply:
It's a case that I can't forsee but I think it's okay to be extra conservative and make sure that it doesn't happen, e.g., given that someone feeds in a custom listing of actions and has accidentally one without field (or a None).